This patch is taken from debian's glibc package (generate-supported.mk).

diff --git a/localedata/Makefile.old b/localedata/Makefile
index 34877c0..a61d9b9 100644
--- a/localedata/Makefile.old
+++ b/localedata/Makefile
@@ -91,7 +91,8 @@ endif
 # Files to install.
 install-others := $(addprefix $(inst_i18ndir)/, \
 			      $(addsuffix .gz, $(charmaps)) \
-			      $(locales))
+			      $(locales)) \
+                   $(inst_i18ndir)/SUPPORTED
 
 tests: $(objdir)/iconvdata/gconv-modules
 
@@ -195,6 +196,14 @@ endif
 
 include SUPPORTED
 
+$(inst_i18ndir)/SUPPORTED: SUPPORTED $(+force)
+	for locale in $(SUPPORTED-LOCALES); do \
+		[ $$locale = true ] && continue; \
+		echo $$locale | sed 's,/, ,' >> LOCALES; \
+	done
+	$(make-target-directory)
+	$(INSTALL_DATA) LOCALES $@
+
 INSTALL-SUPPORTED-LOCALES=$(addprefix install-, $(SUPPORTED-LOCALES))
 
 # Sometimes the whole collection of locale files should be installed.
